File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cs
using System;

public class Solution {
    public int solution(int[] arr) {
        int length = arr.Length;
        int[] temp = new int[length];
        int answer = 0;
        
        while(true)
        {
            int count = length;
            answer++;
            for(int i = 0 ; i < length; i++)
            {
                if(arr[i] > 50 && arr[i] % 2 == 0)
                {
                    temp[i] = arr[i] / 2;
                }
                else if(arr[i] < 50 && arr[i] % 2 != 0)
                {
                    temp[i] = (arr[i] * 2) + 1;
                }
                else
                {
                    temp[i] = arr[i];
                }                   
            }
            for(int i = 0 ; i < length; i++)
            {
                if(temp[i] == arr[i])
                    count--;
                
                arr[i] = temp[i];
            }
            
            if(count == 0)
            {
                answer--;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr) {
    int answer = 0;
    vector<int> prev = arr;
    vector<int> next;
    
    while(true)
    {
        answer++;
        
        for(int i = 0; i < prev.size(); i++)
        {
            if(prev[i] >= 50 && prev[i] % 2 == 0)
                next.push_back(prev[i] % 2);
            else if(prev[i] < 50 && prev[i] % 2 != 0)
                next.push_back(prev[i] * 2 + 1);
            else
                next.push_back(prev[i]);
        }

        for(int i = 0; i < prev.size(); i++)
        {
            if(prev[i] != next[i])
                break;

            if(i == prev.size() - 1 && prev[i] == next[i])
                return answer - 1;
        }
        
        prev = next;
        next.clear();
    }
    
    return answer;
}